streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredKeyValueStore.java

```diff
@@ -115,6 +115,14 @@ public void init(final ProcessorContext context,
 
     @SuppressWarnings("unchecked")
     void initStoreSerde(final ProcessorContext context) {
+        if (context.appConfigs() != null) {
```

Comment:
`appConfigs()` will never return `null` -- we can skip this test IMHO

Reply:
It might also be nice, to do the check if `keySerde != null` only once and avoid a second check when creating `StateSerdes`. Something like:
```
void initStoreSerde(final ProcessorContext context) {
  final Serde<K> usedKeySerde;
    final Serde<V> usedValueSerde;
    if (keySerde == null) {
        usedKeySerde = (Serde<K>) context.keySerde();
     } else {
        usedKeySerde = keySerde;
        usedKeySerde.configure(context.appConfigs(), true);
    }
    // same for valueSerde
    serdes = new StateSerdes<>(
        ProcessorStateManager.storeChangelogTopic(context.applicationId(), name()),
        usedKeySerde,
        usedValueSerde);
```

Reply:
About `appConfigs()` check: is it possible in some unit tests, that mock context is used which could return null? If not, then I agree it is not necessary to check `appConfigs() != null` anywhere.

Reply:
Might be better to update the tests instead?